### PR TITLE
[Feature]: Create class directory component

### PR DESCRIPTION
**Pull request by JuanYin1**
# Class Directory Feature - Complete Implementation

  Core Features Implemented

  - Class Directory Service - Organized member retrieval by roles (professors, TAs, tutors, students)
  - Group Management - Display grouped students with leader/member roles and group details
  - Role-Based Organization - Automatic categorization of class members by their assigned roles
  - Responsive UI Components - Mobile-friendly directory layout with member cards and group displays
  - HTMX Integration - Dynamic content loading with professional styling and animations

  Technical Implementation

  - Database Schema - Full Prisma integration with User, ClassRole, Group, and GroupRole models
  - API Endpoints - Dual endpoints for HTML rendering (/directory) and JSON data (/directory/json)
  - Service Layer - Optimized data fetching with proper relationships and sorting
  - HTML Templating - Rich member cards with contact info, avatars, and role indicators

  Comprehensive Testing Suite

  - BDD Tests - 10 detailed scenarios using Jest-Cucumber framework
  - Coverage - 37 passing tests covering all directory functionality
  - Edge Cases - Empty classes, mixed roles, non-existent classes, and detailed member info
  - Test Infrastructure - Full database setup/teardown with realistic test data

  User Experience Features

  - Member Profiles - Display names, pronouns, contact info, and bios
  - Leadership Indicators - Visual crown badges for group leaders
  - Contact Integration - Direct email, phone, and GitHub links
  - Accessibility - ARIA labels, keyboard navigation, and screen reader support
  - Visual Design - Color-coded role themes and professional card layouts

  Business Value

  - Class Management - Easy overview of all class participants and their roles
  - Team Coordination - Clear group structures with leadership hierarchy
  - Communication - Quick access to member contact information
  - Role Clarity - Visual distinction between professors, TAs, tutors, and students

  # Files Modified/Created

  Backend Implementation

  - ```src/services/class.service.js``` - Added getClassDirectory() service function
  - ```src/controllers/class.controller.js``` - Added JSON API endpoint and HTML rendering controllers
  - ```src/routes/class.routes.js``` - Added /directory and /directory/json routes

  Testing Infrastructure

  - ```features/classDirectory.feature``` - BDD scenarios covering all directory functionality
  - ```tests/class/classDirectory.steps.js``` - Jest-Cucumber step definitions and test helpers

### CODE REVIEW

**Comment by JuanYin1** [comment]
- [x] Email is clickable to send email directly
- [ ] add more contact info to display and able to click and link to third party app
- [ ] Will need to link the box to user profile
- [ ] Display more features

# Current UI
Screenshot 2025-11-18 at 7 19 15 PM
Screenshot 2025-11-18 at 7 19 28 PM
Screenshot 2025-11-18 at 7 19 38 PM
Screenshot 2025-11-18 at 7 19 46 PM
Screenshot 2025-11-18 at 7 19 51 PM

File: code/src/controllers/class.controller.js [hunk 1]
```diff
@@ -374,3 +398,1715 @@ function renderFullPage(content, title = 'My Classes') {
...
+      ${emptyStateHTML}
+      ${professorsHTML}
+      ${tasHTML}
+      ${tutorsHTML}
+      ${groupsHTML}
+      ${ungroupedHTML}
+    </div>
+    
+    <style>
+      /* Directory Styles */
```
Comment (AnniePhan02) [inline code/src/controllers/class.controller.js]: Right now, there are many style CSS blocks which are not reusable and makes the code harder to read. Let's make this modular and move all class directory related css into a separate file (ex /public/css/components/class-directory.css )

File: code/src/controllers/class.controller.js [hunk 2]
```diff
@@ -374,3 +398,1715 @@ function renderFullPage(content, title = 'My Classes') {
...
+      
+      ${contactLinks.length > 0 ? `
+        <div class="group-member-contacts">
+          ${contactLinks.join('')}
+        </div>
+      ` : ''}
+    </div>
+    
+    <style>
+      /* Group Member Item Styles */
```
Comment (AnniePhan02) [inline code/src/controllers/class.controller.js]: should be in css modular file

**Review by AnniePhan02 — APPROVED** [review]

**Review by AnniePhan02 — APPROVED** [review]
